src/gitHubContributorFetcher.ts

```diff
@@ -0,0 +1,102 @@
+import { Octokit } from '@octokit/rest';
+
+export interface GitHubContributor {
+    login?: string | undefined;
+    id?: number | undefined;
+    node_id?: string | undefined;
+    avatar_url?: string | undefined;
+    gravatar_id?: string | null | undefined;
+    url?: string | undefined;
+    html_url?: string | undefined;
+    followers_url?: string | undefined;
+    following_url?: string | undefined;
+    gists_url?: string | undefined;
+    starred_url?: string | undefined;
+    subscriptions_url?: string | undefined;
+    organizations_url?: string | undefined;
+    repos_url?: string | undefined;
+    events_url?: string | undefined;
+    received_events_url?: string | undefined;
+    type?: string | undefined;
+    site_admin?: boolean | undefined;
+    contributions?: number | undefined;
+}
+
+export type RepoOwnerAndName = {
+    owner: string;
+    repo: string;
+};
+
+export interface Contributor {
+    login: string;
+}
+
+export class GitHubContributorFetcher {
+    public async fetchContributors(
+        token: string,
+        repositoryIdentifiers: string[]
+    ): Promise<Map<string, Contributor[]>> {
+        const ownerAndRepos = this.extractOwnerAndRepoNames(repositoryIdentifiers);
+
+        const gitHubContributorsByRepoName = await this.fetchGitHubContributorsByRepoName(
+            token,
+            ownerAndRepos
+        );
+
+        return gitHubContributorsByRepoName;
+    }
+
+    public extractOwnerAndRepoNames(repositoryIdentifiers: string[]): RepoOwnerAndName[] {
+        return repositoryIdentifiers.map((contributorInfo) => {
+            const ownerAndRepo = contributorInfo.split('/');
+            const owner = ownerAndRepo[0];
+            const repo = ownerAndRepo[1];
+            const contributorOwnerAndRepo: RepoOwnerAndName = { owner, repo };
+
+            return contributorOwnerAndRepo;
+        });
+    }
+
+    public async fetchGitHubContributorsByRepoName(
+        token: string,
+        ownerAndRepos: RepoOwnerAndName[]
+    ): Promise<Map<string, Contributor[]>> {
+        const octokit = new Octokit({
+            auth: token,
+        });
+
+        const gitHubContributorsByRepoName = new Map<string, Contributor[]>();
+
+        for (const ownerAndRepoName of ownerAndRepos.values()) {
+            const repoIdentifier = {
```

Comment:
Copying this into a new object looks redundant to me. I might be missing something.